pkg/controllers/repositoryserver/server.go

```diff
@@ -0,0 +1,236 @@
+					NewUsername:  serverUsername,
+					UserPassword: string(password),
+				})
+			stdout, stderr, err := kube.Exec(h.KubeCli, h.RepositoryServer.Namespace, h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, cmd, nil)
+			format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stdout)
+			format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stderr)
+			if err != nil {
+				errors.Wrap(err, "Failed to update existing user passphrase from the Kopia API server")
+			}
+			continue
+		}
+		cmd := command.ServerAddUser(
+			command.ServerAddUserCommandArgs{
+				CommandArgs: &command.CommandArgs{
+					RepoPassword:   repoPassword,
+					ConfigFilePath: command.DefaultConfigFilePath,
+					LogDirectory:   command.DefaultLogDirectory,
+				},
+				NewUsername:  serverUsername,
+				UserPassword: string(password),
+			})
+		stdout, stderr, err := kube.Exec(h.KubeCli, h.RepositoryServer.Namespace, h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, cmd, nil)
+		format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stdout)
+		format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stderr)
+		if err != nil {
+			return errors.Wrap(err, "Failed to add new user to the Kopia API server")
+		}
+	}
+
+	repoServerAddress, serverAdminUserName, serverAdminPassword, err := h.getServerDetails(ctx)
+	if err != nil {
+		return err
+	}
+	err = h.refreshServer(ctx, repoServerAddress, serverAdminUserName, serverAdminPassword)
+	if err != nil {
+		return errors.Wrap(err, "Failed to refresh repository server")
+	}
+	return nil
+}
+
+func (h *RepoServerHandler) refreshServer(ctx context.Context, serverAddress, username, password string) error {
+	repoPassword := string(h.RepositoryServerSecrets.repositoryPassword.Data[repoPasswordKey])
+	fingerprint, err := kopia.ExtractFingerprintFromCertSecret(ctx, h.KubeCli, h.RepositoryServerSecrets.serverTLS.Name, h.RepositoryServer.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "Failed to extract fingerprint Kopia API Server Certificate Secret Data")
+	}
+
+	cmd := command.ServerRefresh(
+		command.ServerRefreshCommandArgs{
+			CommandArgs: &command.CommandArgs{
+				RepoPassword:   repoPassword,
+				ConfigFilePath: command.DefaultConfigFilePath,
+				LogDirectory:   command.DefaultLogDirectory,
+			},
+			ServerAddress:  serverAddress,
+			ServerUsername: username,
+			ServerPassword: password,
+			Fingerprint:    fingerprint,
+		})
+	stdout, stderr, err := kube.Exec(h.KubeCli, h.RepositoryServer.Namespace, h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, cmd, nil)
+	format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stdout)
+	format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stderr)
+	if err != nil {
+		errors.Wrap(err, "Failed to refresh Kopia API server")
```

Comment:
As pointed to above, the string there says `Failed to refresh repository server`